### ISSUE TITLE
Total Steps when split_batches = True?

### ISSUE THREAD

**Issue by karan-dalal** [huggingface/accelerate#2708]
### System Info

```Shell
- `Accelerate` version: 0.29.3
- Platform: Linux-5.15.0-41-generic-x86_64-with-glibc2.31
- `accelerate` bash location: /nlp/scr/yusun/miniconda3/envs/mamba/bin/accelerate
- Python version: 3.11.9
- Numpy version: 1.26.4
- PyTorch version (GPU?): 2.2.0+cu121 (True)
- PyTorch XPU available: False
- PyTorch NPU available: False
- PyTorch MLU available: False
- System RAM: 1007.69 GB
- GPU type: NVIDIA A100-SXM4-80GB
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: MULTI_GPU
        - mixed_precision: bf16
        - use_cpu: False
        - debug: False
        - num_processes: 4
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: all
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - enable_cpu_affinity: False
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
```

### Information

- [x] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

```python
import os
import argparse
import torch
import torch.optim as optim
from torch.nn.functional import cross_entropy

from loader.pile.language_modeling_hf import LMDataModule

from tqdm import tqdm
from accelerate import Accelerator
from accelerate.utils import set_seed
from mamba_ssm.models.mixer_seq_simple import MambaLMHeadModel

from transformers import MambaConfig
from transformers.optimization import get_cosine_with_min_lr_schedule_with_warmup

def save_args(args, experiment_dir):
    args_dict = vars(args)
    args_path = os.path.join(experiment_dir, 'args.txt')
    with open(args_path, 'w') as f:
        for key, value in args_dict.items():
            f.write(f"{key}: {value}\n")

def run(args):
    set_seed(42)

    experiment_dir = os.path.join(args.base_dir, 'exp', args.experiment_name)
    accelerator = Accelerator(mixed_precision='bf16', split_batches=True, gradient_accumulation_steps=args.grad_accum, project_dir=experiment_dir)
    device = accelerator.device
    
    data_module = LMDataModule(dataset_name=args.dataset,
                               dataset_config_name=None,
                               tokenizer_name='EleutherAI/gpt-neox-20B',
                               cache_dir=args.cache_dir,
                               max_length=args.seq_len,
                               add_eos=True,
                               batch_size=args.global_batch_size,
                               batch_size_eval=args.global_batch_size,
                               num_workers=args.num_workers,
                               use_shmem=False,
                               shuffle=True,
                               fault_tolerant=True,
                               ddp=False,
                               drop_last=True)
                        
    data_module.prepare_data()
    data_module.setup()
    train_dataloader = data_module.train_dataloader()

    config = MambaConfig.from_pretrained(args.config)
    config.tie_embeddings = True
    model = MambaLMHeadModel(config, device=device)

    optimizer = optim.AdamW(model.parameters(),
                            lr=args.learning_rate,
                            weight_decay=0.1,
                            betas=(0.9, 0.95))
    scheduler = get_cosine_with_min_lr_schedule_with_warmup(optimizer, 
                                                            num_warmup_steps=args.warmup_steps * accelerator.num_processes,
                                                            num_training_steps=args.total_steps * accelerator.num_processes,
                                                            min_lr=args.min_learning_rate)

    model, optimizer, train_dataloader, scheduler = accelerator.prepare(
        model, optimizer, train_dataloader, scheduler
    )

    if accelerator.is_local_main_process:
        os.makedirs(experiment_dir, exist_ok=True)
        save_args(args, experiment_dir)
        all_stat_dict = {'train/loss': [], 'learning_rate': []}
        train_loss, lr = [], []
        print(f"Beginning Training. Experiment will be saved in {experiment_dir}")

    avg_loss, current_step = 0.0, 0
    pbar = tqdm(total=args.total_steps, desc="Training", disable=not accelerator.is_local_main_process)

    for batch in train_dataloader:
        if current_step >= args.total_steps:
                break
        
        with accelerator.accumulate(model):
            optimizer.zero_grad()

            inputs, targets, loss_mask = batch['input_tokens'], batch['target_tokens'], batch['loss_masks']
            targets = targets.long()

            outputs = model(inputs)
            logits = outputs.logits

            loss = cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1), reduction='none')
            loss = (loss * loss_mask.view(-1)).mean()
            avg_loss += loss.item()

            accelerator.backward(loss)
            optimizer.step()
            scheduler.step()

        if accelerator.sync_gradients:
            current_step += 1
            avg_loss /= accelerator.gradient_accumulation_steps

            if accelerator.is_local_main_process:
                pbar.update(1)
                pbar.set_postfix(loss=avg_loss)

                train_loss.append(avg_loss)
                lr.append(scheduler.get_last_lr()[0])
                avg_loss = 0.0

                if current_step % args.log_freq == 0:
                    all_stat_dict['train/loss'].extend(train_loss)
                    all_stat_dict['learning_rate'].extend(lr)
                    torch.save(all_stat_dict, os.path.join(experiment_dir, 'all_stat_dict.pth'))
                    train_loss, lr = [], []

            if current_step % args.save_freq == 0:
                accelerator.wait_for_everyone()
                accelerator.save_state(experiment_dir + '/checkpoint/' + str(current_step))

    if accelerator.is_local_main_process:
        pbar.close()
        print("Training Completed")
    accelerator.end_training()

if __name__ == "__main__":
    parser = argparse.ArgumentParser()

    parser.add_argument("--config", type=str, default="state-spaces/mamba-130m-hf")
    parser.add_argument("--dataset", type=str, default="the_pile")
    parser.add_argument("--cache_dir", type=str, default="the_pile")

    parser.add_argument("--num_workers", type=int, default=2)
    parser.add_argument("--seq_len", type=int, default=2048)
    parser.add_argument("--global_batch_size", type=int, default=32)
    
    parser.add_argument("--learning_rate", type=float, default=3e-3)
    parser.add_argument("--min_learning_rate", type=float, default=1e-5)
    parser.add_argument("--warmup_steps", type=int, default=3)
    parser.add_argument("--total_steps", type=int, default=30)
    parser.add_argument("--grad_accum", type=int, default=1)

    parser.add_argument("--base_dir", type=str, default="mamba")
    parser.add_argument("--experiment_name", type=str, default="debug")
    parser.add_argument("--log_freq", type=int, default=3)
    parser.add_argument("--save_freq", type=int, default=1000)

    args = parser.parse_args()
    run(args)
``` 

### Expected behavior

Hi. I'm trying to understand what behavior I should use for my scheduler.

In the docs (https://huggingface.co/docs/accelerate/en/concept_guides/performance#learning-rates) they recommend multiplying the total number of warmup/steps by the number of processes.

In the above case, I log my learning rate. When visualized:
<img width="978" alt="Screenshot 2024-04-24 at 2 28 26 PM" src="https://github.com/huggingface/accelerate/assets/44450861/69b63cc4-6307-4bb6-8d63-fd13b64aa485">

This seems to actually 4x the total number of steps. VS when I remove the ```accelerator.num_processes```:
<img width="974" alt="Screenshot 2024-04-24 at 2 29 20 PM" src="https://github.com/huggingface/accelerate/assets/44450861/bf7ff2ad-d4ef-47c5-a7a4-3bfad5724c9a">

Which is the expected behavior. 

I feel it has something to do with using ```split_batches=True``` (https://github.com/huggingface/accelerate/blob/e831bcb3b1bc01b517f8d58987cba466bdc19bbf/src/accelerate/scheduler.py#L69).

It was a little hard for me to parse the docs on this; **it would be great if I could get the recommended way to create the scheduler so it interacts correctly with Accelerate**. I'm looking to define the global batch size for all devices in my ```args``` as well as the total steps using this global batch size. As you can also see from the code, I eventually want to use gradient accumulation to increase this global batch size.

Happy to make a PR to update docs after.
